packages/@lwc/template-compiler/README.md

```diff
@@ -51,6 +51,38 @@ const { code, warnings } = compile(`<template><h1>Hello World!</h1></template>`,
 -   `experimentalComputedMemberExpression` (boolean, optional, `false` by default) - set to `true` to enable computed member expression in the template, eg: `{list[0].name}`.
 -   `experimentalDynamicDirective` (boolean, optional, `false` by default) - set to `true` to allow the usage of `lwc:dynamic` directives in the template.
 -   `preserveHtmlComments` (boolean, optional, `false` by default) - set to `true` to disable the default behavior of stripping HTML comments.
+-   `customRendererConfig` (CustomRendererConfig, optional) - specifies a configuration to use to match elements. Matched elements will get a custom renderer hook in the generated template.
+
+    -   Example 1: Config to match `<use>` elements under the `svg` namespace and have `href` attribute set.
+
+        ```
+        {
+            customRendererConfig: {
+                directives: [],
+                elements: [
+                    {
+                        tagName: 'use',
+                        namespace: 'http://www.w3.org/2000/svg',
+                        attributes: ['href']
+                    }
+                ]
+            }
+        }
+        ```
+
+    -   Example 2: Config to match `<script>` elements regardless of the attribute set. _Note:_ When `attributes` is not specified, attribute matching is skipped.
+        ```
+        {
+            customRendererConfig: {
+                directives: [],
+                elements: [
+                    {
+                        tagName: 'script'
+                    }
+                ]
+            }
+        }
+        ```
```

Comment:
Chatted with Caridy about this, and he pointed out that even if the declarative format didn't live in LWC OSS, it would probably live in `lwc-platform`. So either way, we are going to own it. In that case, I'm fine with keeping it here, although in the future I could see refactoring it into `lwc-platform`. It's more of a platform concern anyway (e.g. need a declarative format for something interoperable between Java/Node).